src/quiz/use-quiz.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+import { type QuizQuestionProps } from "../quiz-question";
+
+export const useQuiz = (initialValues = {}) => {
+	const [values, setValues] =
+		useState<Record<string, QuizQuestionProps["selectedOption"]>>(
+			initialValues,
+		);
+
+	const handleChange: QuizQuestionProps["onChange"] = ({
+		questionId,
+		selectedOption,
+	}) => {
+		setValues({
+			...values,
+			[questionId]: selectedOption,
+		});
+	};
+
+	return { values, handleChange };
```

Comment:
I'm not sure about using the question text as a `questionId`. It seems a bit sketchy to use something meant for user consumption as an id for machines. Also, it means we have to explain what it is!

How about this alternative: have `useQuiz` represent the values (answers? options?) as an array and return an array of `[value, setValue]` pairs? Then you can just pass those along to the `QuizQuestion` rather than it need to "decide" which value to update. It's more modular: each `QuizQuestion` only knows about its own questions and answers.

The other nice feature is that it puts questions and answers on the same footing. They're both arrays rather than one being an array and the other being a map.